### PR TITLE
Change to compile jemalloc from source

### PR DESCRIPTION
**Pull request by gaffneyc**
As brought up in #27 there are possible supply chain attacks against
anyone using this buildpack. The main concern is that a compromised
binary could be uploaded and there is no way to check that the hosted
binaries are trusted. Going a level deeper there is no way to verify
that the compiled binaries I've uploaded can be trusted (I promise
they're fine!).

This goes a step farther and removes one more random dude on the
Internet (me!) from the chain of trust. By compiling from source and
posting known checksums of those sources it is possible to verify
exactly what is being built and to be notified if the source changes.
This has the added benefit of not needing to provide binaries for each
new release of Jemalloc or launch of a new Heroku stack. Builds are now
cached per version and stack so that changing either will cause a
rebuild on the first deploy.

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -7,6 +7,21 @@ BUILD_DIR=$1
...
+  ['4.4.0']='a7aea63e9718d2f1adf81d87e3df3cb1b58deb86fc77bad5d702c4c59687b033'
+  ['4.5.0']='9409d85664b4f135b77518b0b118c549009dc10f6cba14557d170476611f6780'
+  ['5.0.1']='4814781d395b0ef093b21a08e8e6e0bd3dab8762f9935bbfb71679b0dea7c3e9'
+  ['5.1.0']='5396e61cc6103ac393136c309fae09e44d74743c86f90e266948c50f3dbb7268'
+  ['5.2.0']='74be9f44a60d2a99398e706baa921e4efde82bf8fd16e5c0643c375c5851e3b4'
+  ['5.2.1']='34330e5ce276099e2e8950d9335db5a875689a4c6a56751ef3b1d8c537f887f6'
+)
+
 # Default version
 version="5.2.1"
```
Comment (pirj) [inline bin/compile:26]: Is it possible to override this somehow? Via Heroku env vars?
Maybe it would be possible to supply a checksum there, too?

**Comment by pirj** [comment]
>  Decide what to do if the checksum is not in the list (e.g. new version comes out but the buildpack hasn't been updated)

Maybe use the latest jemalloc tag that the buildpack is aware of? There's a default version anyway:
```
version="5.2.1"
```

And warn during the build that a new version is available (but who looks at the logs anyway) to update the buildpack?

BTW, [5.3 may be released sometimes soon](https://github.com/jemalloc/jemalloc/issues/2213) 🎉

File: bin/compile (hunk 1 above)
Comment (gaffneyc) [reply bin/compile:26]: Yep, you can override it by setting `JEMALLOC_CHECKSUM`

https://github.com/gaffneyc/heroku-buildpack-jemalloc/pull/28/files#diff-a64a7d9b82578ca3053416d56631a61e0ceabed87619415663030e3842429e05R55

**Comment by gaffneyc** [comment]
> Decide what to do if the checksum is not in the list (e.g. new version comes out but the buildpack hasn't been updated)

I think we can fail the build if `JEMALLOC_VERSION` isn't in the checksum list and `JEMALLOC_CHECKSUM` isn't set. Since an operator is changing the configuration to specify a version of jemalloc to use I think failing the build wouldn't be that surprising since they've now created a configuration problem.

**Comment by gaffneyc** [comment]
Haven't had a chance to circle back on this but we've been using it for months in production. With the [heroku-22](https://devcenter.heroku.com/changelog-items/2429) stack becoming generally available I'm planning to test a stack upgrade then come back and get this finished and merged. I've cut builds for heroku-22 in the mean time.

**Pull request merged**